Trustroots/trustroots
# Admin panel: add role controls

Add better role controls for admins.

See https://github.com/Trustroots/trustroots/issues/1184 for higher level details.

#### Proposed Changes

- Allow admins to add a shadowban role to a user
- Allow admins to add a moderator role to a user
- Allow admins to list users by role


#### Testing Instructions

* Have a user with `admin` in their `roles`
* At `http://localhost:3000/admin/search-users` you can now choose to list users by role
    <img width="388" alt="Screenshot 2020-01-20 at 00 19 26" src="https://user-images.githubusercontent.com/87168/72689534-fee79600-3b1a-11ea-8e11-a74184fdf8ec.png">
* At the listing you should see "moderator" and "shadowban" colored green and red
    <img width="517" alt="Screenshot 2020-01-20 at 00 19 41" src="https://user-images.githubusercontent.com/87168/72689543-16268380-3b1b-11ea-91f3-f2339b7a728c.png">
* At member report card you can add these roles to any user. When pressing, it should ask confirmation if you really wanna do it.
    <img width="1108" alt="Screenshot 2020-01-20 at 00 19 57" src="https://user-images.githubusercontent.com/87168/72689552-1de62800-3b1b-11ea-9fe4-ec769a2f2e87.png">
    <img width="279" alt="image" src="https://user-images.githubusercontent.com/87168/72689556-30f8f800-3b1b-11ea-82a5-ed791f6bba9e.png">

Ideally banning (red) and promoting (green) actions wouldn't be in one button row to avoid accidents but let's just be careful shall we. ;-)
